source/sav/Sav1.cpp

```diff
@@ -172,25 +172,12 @@ namespace pksm
     // why did they use BCD
     u32 Sav1::money() const
     {
-        u32 bcdValue = (BigEndian::convertTo<u16>(&data[OFS_MONEY]) << 8) |
-                       data[OFS_MONEY + 2];
-        return (((bcdValue & 0xF00000) >> 20) * 100000) + (((bcdValue & 0x0F0000) >> 16) * 10000) +
-               (((bcdValue & 0x00F000) >> 12) * 1000) + (((bcdValue & 0x000F00) >> 8) * 100) +
-               (((bcdValue & 0x0000F0) >> 4) * 10) + (bcdValue & 0x00000F);
+        return NumberFormatUtils::BigEndianBCDtoUInteger<u32>(&data[OFS_MONEY], 3);
     }
     void Sav1::money(u32 v)
     {
-        u32 result = 0;
-        if (v > 999999) v = 999999;
-        result |= (v / 100000) << 20;
-        result |= ((v / 10000) % 10) << 16;
-        result |= ((v / 1000) % 10) << 12;
-        result |= ((v / 100) % 10) << 8;
-        result |= ((v / 10) % 10) << 4;
-        result |= v % 10;
-        BigEndian::convertFrom<u16>(
-            &data[OFS_MONEY], u16((result & 0xFFFF00) >> 8));
-        data[OFS_MONEY + 2] = u8(result & 0x0000FF);
+        std::array<u8, 3> bcd = NumberFormatUtils::UIntegerToBigEndianBCD<u32, 3>(v);
```

Comment:
Use the change to this function requested to write directly instead of having a temporary